reference-delegate-explainer.md

```diff
@@ -0,0 +1,534 @@
+<script>
+  const fancyInput = document.getElementById('fancy-input');
+  const realInput = fancyInput.shadowRoot.getElementById('real-input');
+
+  console.log(realInput.labels); 
+  // [<label id="outer-label">, <label id="inner-label">]
+
+  console.log(fancyInput.labels);
+  // undefined (`labels` is not an attribute of HTMLElement)
+  // If fancyInput were a form-associated custom element, then elementInternals.labels
+  // would return an empty list [] because the labels are delegated to realInput.
+</script>
+```
+
+#### Interaction with CSS Selectors
+
+The referenceDelegate has no interaction with CSS. An ID selector will target the host element, and _not_ its referenceDelegate.
+
+## Privacy and Security Considerations
+
+No considerable privacy or security concerns are expected, but community feedback is welcome.
+
+## Considered Alternatives
+
+This section covers some design alternatives, along with discussion of their Pros and Cons, and why they were not included in the design.
+
+### Alternative names for the feature "Reference Delegate"
+
+The name "reference delegate" (`shadowrootreferencedelegate`) is intended to indicate that the specified element acts as the delegate for references to the host element. Some possible alternative names:
+* "Delegates References" - `shadowrootdelegatesreferences="id"` - more similar wording to `shadowrootdelegatesfocus`.
+* "Reflects References" - `shadowrootreflectsreferences="id"` - borrowing from the [Cross-root ARIA Reflection API](https://github.com/Westbrook/cross-root-aria-reflection/blob/main/cross-root-aria-reflection.md) propsal.
+* "Forwards References" - `shadowrootforwardsreferences="id"` - borrowing from ["forwardRef" in React](https://react.dev/reference/react/forwardRef).
+
+Ultimately, the name "reference delegate" is the most concise and conveys the intent of the feature. However, community feedback is welcome on the name.
+
+### Omit the "catch-all" `referenceDelegate` attribute
+
+It is technically possible to require all attributes to be individually delegated via `referenceDelegateMap`, rather than also allowing `referenceDelegate` as a "catch-all" for every attribute.
+
+#### Pros
+
+* The main argument to omit `referenceDelegate` is that the semantics could change if more delegated attributes are added in the future. This could break existing websites by changing the target of an attribute, if it is added to `referenceDelegate` support in the future.
+* It makes it more difficult for browser vendors to incrementally implement reference delegate, since adding support for additional attributes is a breaking change.
+
+#### Cons
+
+* The Reference Delegate feature is intended to support all attributes that use ID references. Thus, the only time a new attribute will be supported by Reference Delegate is when it is a completely new attribute in the HTML spec. There is no backwards compatibility concern, since no websites will be using the new attribute before is is supported.
```

Comment:
I think this argument works when scoped to targeting, although presumably it should not be limited to ID references. What about `<label>`-sans-`for=""`?

However, before there was a suggestion that all kinds of other behavioral things would be delegated as well. E.g., focus would magically transfer. I guess that's off the table now?

Reply:
There's a [Nesting inside `<label>`](https://github.com/behowell/aom/blob/reference-delegate/reference-delegate-explainer.md#nesting-inside-label) section that currently says it'll work when nested in a label.

However, as I mentioned in a note in that section, that is the only part of this feature that doesn't directly work with IDREF attributes. Perhaps it's better to keep `referenceDelegate` scoped to deal _only_ with ID references?

Delegating other things isn't off the table; but it would be out of scope of this particular feature. Previous proposals have stalled in part due to scope creep, and it might be best to keep this feature focused on the one piece currently blocking ARIA from working across shadow roots.

Reply:
I'm reluctantly in favour of leaving `<label>` wrapped out of scope - as much as I would love it to work, I agree that it feels like scope creep. 

It does seem like if it's included, it should be an explicit inclusion, to avoid opening a can of worms in figuring out how all the other types of nesting element behaviours would work and/or what would be in/out of scope.

Reply:
I think `<label>` without `for` should work. I don't know if any other elements that work like it, but I guess we should double check that.

Reply:
I've updated the proposal to say that label wrapping doesn't work without the `for` attribute. This keeps the feature focused only on ID references. I'm open to changing it back if there is strong feedback in favor of it.